services/gcn_service/gcn_service.py

```diff
@@ -15,16 +17,49 @@
     getattr(gcn.notice_types, notice_type) for notice_type in cfg["gcn_notice_types"]
 ]
 
+gcn_observation_plans = [
+    observation_plan for observation_plan in cfg["gcn_observation_plans"]
+]
+
+log = make_log('gcnserver')
+
 
 def handle(payload, root):
     notice_type = gcn.get_notice_type(root)
     if notice_type in notice_types:
         user_id = 1
         with DBSession() as session:
-            post_gcnevent(payload, user_id, session)
+            event_id = post_gcnevent(payload, user_id, session)
 
+            start_date = str(datetime.utcnow()).replace("T", "")
+            end_date = str(datetime.utcnow() + timedelta(days=1)).replace("T", "")
+            for gcn_observation_plan in gcn_observation_plans:
+                proposal_id = gcn_observation_plan['allocation-proposal_id']
+                allocation = (
+                    session.query(Allocation)
+                    .filter(Allocation.proposal_id == proposal_id)
+                    .first()
+                )
+                event = session.query(GcnEvent).get(event_id)
 
-if __name__ == "__main__":
-    log = make_log("gcnserver")
+                if allocation is not None:
```

Comment:
Maybe you could query the GcnEvents after the if statement that checks if there is no allocation? Just a small detail but it avoids querying the gcn events if there is no allocation in the first place.

Reply:
I moved it up just after the event is committed so it is only loaded once actually, which I think is better. What do you think?